tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/telemetryMetrics/util/TelemetryHistogramMetric.java

```diff
@@ -32,31 +42,126 @@
  * Most tests should use {@link TelemetryMetricGetter} to create instances of this class.
  */
 public class TelemetryHistogramMetric {
+
+    public static final List<Double> EXPECTED_BOUNDARIES = Arrays.asList(0.005, 0.01, 0.025, 0.05, 0.075, 0.1,
+            0.25, 0.5, 0.75, 1d, 2.5, 5d, 7.5, 10d);
     private final TelemetryMetricID metricId;
+    private final double MINIMUM_UPPER_BOUND_SECONDS;
 
     public TelemetryHistogramMetric(TelemetryMetricID metricId) {
         this.metricId = metricId;
+        MINIMUM_UPPER_BOUND_SECONDS = TCKConfig.getConfig().getTimeoutInMillis(200) / 1_000d;
     }
 
-    @SuppressWarnings("unchecked") // Fail early on a misconfigured test
-    public Collection<HistogramPointData> getHistogramPoints() {
+    public Optional<HistogramPointData> getHistogramPoint() {
         InMemoryMetricReader reader = InMemoryMetricReader.current();
-        return (List<HistogramPointData>) reader.getPointData(metricId);
+        return reader.getMetric(metricId)
+                .flatMap(md -> InMemoryMetricReader.getHistogramPointData(md, metricId));
     }
 
     public Optional<Long> getHistogramCount() {
-        Collection<HistogramPointData> hpd = getHistogramPoints();
+        return getHistogramPoint()
+                .map(HistogramPointData::getCount);
+    }
+
+    public boolean isPresent() {
+        return getHistogramPoint().isPresent();
+    }
+
+    public void assertBoundaries() {
+        HistogramPointData pointData =
+                getHistogramPoint().orElseThrow(() -> new AssertionError("No data point for " + metricId.name));
+        assertThat(pointData.getBoundaries(), equalTo(EXPECTED_BOUNDARIES));
+    }
 
-        if (hpd.isEmpty()) {
-            return Optional.empty();
+    /**
+     * Assert that the bucket counts of the histogram are consistent with the expected set of recorded results.
+     * <p>
+     * The results are adjusted using {@link TCKConfig#getTimeoutInMillis(long)}.
+     * <p>
+     * A 20% deviation from the expected time is permitted and the upper limit is extended to 200ms.
+     *
+     * @param expectedResultsMillis
+     *            the list of values which the histogram is expected to have recorded
+     */
+    public void assertBucketCounts(long... expectedResultsMillis) {
+        HistogramPointData pointData =
+                getHistogramPoint().orElseThrow(() -> new AssertionError("No data point for " + metricId.name));
+
+        // Check we have the right number of results in the histogram:
+        assertEquals(pointData.getCount(), expectedResultsMillis.length,
+                "Wrong number of results in histogram getCounts for " + metricId.name);
```

Comment:
Please can we have metricId.toString() for these three asserts.

Reply:
Done